File: runtimes/kotlin-1.8/src/main/kotlin/io/openruntimes/kotlin/RuntimeContext.kt
package io.openruntimes.kotlin

class RuntimeContext(val req: RuntimeRequest, val res: RuntimeResponse, val logger: RuntimeLogger) {
    fun log(message: Any) {
        this.logger.write(message + "\n", RuntimeLogger.TYPE_LOG)
    }

    fun error(message: Any) {
        this.logger.write(message + "\n", RuntimeLogger.TYPE_ERROR)
    }
}

File: runtimes/dotnet/versions/latest/src/RuntimeContext.cs
namespace DotNetRuntime
{
    public class RuntimeContext
    {
        public RuntimeRequest Req { get; set; }
        public RuntimeResponse Res { get; set; }
        public RuntimeLogger Logger { get; set; }

        public RuntimeContext(RuntimeRequest req, RuntimeResponse res, RuntimeLogger logger)
        {
            Req = req;
            Res = res;
            Logger = logger;
        }

        public void Log(object message)
        {
            this.Logger.Write(message, RuntimeLogger.TYPE_LOG);
        }

        public void Error(object message)
        {
            this.Logger.Write(message, RuntimeLogger.TYPE_ERROR);
        }
    }
}
